packages/office-ui-fabric-react/src/components/Layer/examples/Layer.Basic.Example.tsx

```diff
@@ -1,80 +1,60 @@
-
-interface ILayerContentExampleState {
-  time: string;
-}
-
-class LayerContentExample extends BaseComponent<{}, ILayerContentExampleState> {
-  public state = {
-    time: new Date().toLocaleTimeString()
-  };
-
-  public componentDidMount() {
-    this._async.setInterval(() => {
-      this.setState({
-        time: new Date().toLocaleTimeString()
-      });
-    }, 1000);
-  }
-
-  public render() {
-    return (
-      <LayerBasicExampleContext.Consumer>
-        {value => (
-          <div className={css(styles.content, AnimationClassNames.scaleUpIn100)}>
-            <div className={styles.textContent}>{value.message}</div>
-            <div>{this.state.time}</div>
-          </div>
-        )}
-      </LayerBasicExampleContext.Consumer>
-    );
+// tslint:disable:jsx-no-lambda
+
+const theme = getTheme();
+const classNames = mergeStyleSets({
+  content: {
+    backgroundColor: theme.palette.themePrimary,
+    color: theme.palette.white,
+    lineHeight: '50px',
+    padding: '0 20px',
+    display: 'flex'
+  },
+  textContent: {
+    flexGrow: '1'
   }
-}
-
-interface ILayerBasicExampleState {
-  showLayer: boolean;
-}
-
-export class LayerBasicExample extends BaseComponent<{}, ILayerBasicExampleState> {
-  public state = {
-    showLayer: false
-  };
-
-  public render() {
-    const { showLayer } = this.state;
-    return (
-      <LayerBasicExampleContext.Provider
-        value={{
-          message: 'Hello world.'
-        }}
-      >
-        <div>
-          <Toggle label="Wrap the content box below in a Layer" inlineLabel checked={showLayer} onChange={this._onChange} />
-
-          {showLayer ? (
-            <Layer>
-              <LayerContentExample />
-            </Layer>
-          ) : (
-            <LayerContentExample />
-          )}
-        </div>
-      </LayerBasicExampleContext.Provider>
-    );
-  }
-
-  private _onChange = (ev: React.FormEvent<HTMLElement | HTMLInputElement>, checked: boolean): void => {
-    this.setState({ showLayer: checked });
-  };
-}
+});
+
+const LayerContentExample: React.FunctionComponent = () => {
+  const [time, setTime] = React.useState<string>(new Date().toLocaleTimeString());
+
+  React.useEffect(() => {
+    const interval = setInterval(() => setTime(new Date().toLocaleTimeString()));
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <div className={css(classNames.content, AnimationClassNames.scaleUpIn100)}>
+      <div className={classNames.textContent}>Hello world</div>
+      <div>{time}</div>
+    </div>
+  );
+};
+
+export const LayerBasicExample: React.FunctionComponent = () => {
+  const [showLayer, setShowLayer] = React.useState<boolean>(false);
+
+  return (
+    <div>
+      <Toggle
+        label="Wrap the content box below in a Layer"
+        inlineLabel
+        checked={showLayer}
+        onChange={(ev: any, checked?: boolean) => setShowLayer(!!checked)}
```

Comment:
For callbacks:

```tsx
const onChange = useCallback(..., [ ])
```

Reply:
Do you know how important it actually is perf-wise to use `useCallback`?

Reply:
My thought was that not using it would cause the callback to mutate every time, possibly breaking a memo'd component contract. It seems like a habit to encourage over constantly mutating callbacks.

Reply:
The thing is that we're already breaking memoized components any time we pass an inline-created object as a prop, which happens quite often (and we don't have a lint rule against it). So I'm not convinced that passing callbacks this way is much different.

Reply:
We often choose conciseness of code when doing examples and helping others (I often inline objects myself) but should we set a higher standard for our examples?

In general inline created props should be discouraged and I don't think we should be using them in our examples. People often copy and paste and now bad patterns are in their code. We should consider enabling a lint rule against them. This will be particularly important if we start considering immutable props our base requirement going forward.

In this case `onChange` is really only dependent on `checked`, which makes it a prime candidate for `useMemo`.